src/rank_llm/rerank/rankllm.py

```diff
@@ -143,6 +143,53 @@ def permutation_pipeline(
         result = self.receive_permutation(result, permutation, rank_start, rank_end)
         return result
 
+    def permutation_pipeline_batched(
+        self,
+        results: List[Result],
+        rank_start: int,
+        rank_end: int,
+        logging: bool = False,
+    ) -> List[Result]:
+        """
+        Runs the permutation pipeline on the passed in result set within the passed in rank range for a batch of results.
+
+        Args:
+            results (List[Result]): The list of result objects to process.
+            rank_start (int): The start index for ranking.
+            rank_end (int): The end index for ranking.
+            logging (bool, optional): Flag to enable logging of operations. Defaults to False.
+
+        Returns:
+            List[Result]: The processed list of result objects after applying permutation.
+        """
+        prompts = []
+        for result in results:
+            prompt, in_token_count = self.create_prompt(result, rank_start, rank_end)
+            if logging:
+                print(f"prompt: {prompt}\n")
+            prompts.append((prompt, in_token_count))
```

Comment:
Nit: you don't need to store prompt and in token count as a tuple here and later on zip the list of tuples with reuslts to read them and create rankingExecInfo.
I'd say just create the list of rankingExecInfos with input info and edit the objects to add output info in the follow up for loop